gobblin-service/src/test/java/org/apache/gobblin/service/modules/orchestration/MostlyMySqlDagManagementStateStoreTest.java

```diff
@@ -98,12 +98,7 @@ public void testAddDag() throws Exception {
     Assert.assertEquals(dag.toString(), this.dagManagementStateStore.getDag(dagId).get().toString());
     Assert.assertEquals(dagNode, this.dagManagementStateStore.getDagNodeWithJobStatus(dagNodeId).getLeft().get());
 
-    List<Dag.DagNode<JobExecutionPlan>> dagNodes = this.dagManagementStateStore.getDagNodes(dagId);
-    Assert.assertEquals(2, dagNodes.size());
-    Assert.assertEquals(dagNode, dagNodes.get(0));
-    Assert.assertEquals(dagNode2, dagNodes.get(1));
-
-    dagNodes = this.dagManagementStateStore.getDagNodes(dagId);
+    Set<Dag.DagNode<JobExecutionPlan>> dagNodes = this.dagManagementStateStore.getDagNodes(dagId);
     Assert.assertEquals(2, dagNodes.size());
     Assert.assertTrue(dagNodes.contains(dagNode));
     Assert.assertTrue(dagNodes.contains(dagNode2));
```

Comment:
do the contains checks here verify the new equals/hashcode is working?

Reply:
added an assert separately for that